internal/rsl/rsl.go

```diff
@@ -442,6 +562,105 @@ func GetLatestReferenceEntryForRefBefore(repo *gitinterface.Repository, refName
 	return targetEntry, annotations, nil
 }
 
+func GetLatestReferenceEntryForRefUntilNumber(repo *gitinterface.Repository, refName string, until uint64) (*ReferenceEntry, []*AnnotationEntry, error) {
+	return GetLatestReferenceEntryForRefBeforeUntilNumber(repo, refName, gitinterface.ZeroHash, until)
+}
+
+// GetLatestReferenceEntryForRefBeforeUntilNumber returns the latest reference
+// entry available locally in the RSL for the specified refName before the
+// specified anchor. The search is terminated when an entry with number less
+// than the until value passed in is encountered. Note that until is inclusive:
+// if the entry in question is for the requested reference, it is returned. If
+// the latest entry is not numbered (e.g., in a repository that predates this
+// feature), this function is identical to GetLatestReferenceEntryForRefBefore.
+func GetLatestReferenceEntryForRefBeforeUntilNumber(repo *gitinterface.Repository, refName string, anchor gitinterface.Hash, until uint64) (*ReferenceEntry, []*AnnotationEntry, error) {
```

Comment:
Could this func name be a bit smaller/simplified?

Reply:
Probably. Do you have any suggestions?

Reply:
Could we change the UntilNumber to Anchor, or somthing else?

Reply:
Anchor means an entry ID (ie its commit ID) rather than number, which is why I was trying to avoid confusion.

Reply:
bikeshed: I'd rename this to GetLatestReferenceEntryBeforeNumber and rename until to number or n or something similar.

- `ReferenceEntryForRef` is redundant
- `until uint64` is going to look like an unix time out of context.

If you're worried about confusion with GetLatestReferenceEntryForRefUntilNumber make this variadic with options and do:

```go
GetLatestReferenceEntry(repo, ref, option.WithAnchor(hash), option.BeforeNumber(n))
```

This would let us cut down on a significant amount of the copy+pasta in this file.

Reply:
we should open a ticket for the completely untested conditions but also we probably want to test for combinations of conditions. I think the combinations we use currently are tested, luckily.